rfd/0001-general-design.md

```diff
@@ -0,0 +1,433 @@
+selector capabilities to route the traffic for those records
+- **DDoS**: As we scale balancer using the centralized storage, and balancer considered to be
+ephemeral in context of multiple-replicas — we can lose replicas and create new replicas on the go
+
+#### Client Service mTLS Layer
+Proposed mTLS layer consists of the following scheme:
+```
+                                                                    ╔════════════╗
+                                                      ┌─────────────╣ Connection ╠─────────────┐
+                                                      │             ║  manager   ║             │
+                                                      │             ╚════════════╝             │
+                                                      │                                        │
+┌──────────┐                                          │       ┌───────────────┐                │
+│          │                                       ┌──┼──────▶│  Accept Cert  │                │
+│  CACert  │───────┐        ┌────────────┐         │  │       └───────┬───────┘                │
+│          │       │        │            │         │  │               │                        │
+└──────────┘  ┌────┴──┐     │   Client   │  ┌──────┴┐ │       ┌───────▼───────┐                │
+              │Produce│     │    Cert    ├──┤Connect│ │       │  Fetch CN     │                │
+┌──────────┐  │  TLS  ├────▶│            │  └───────┘ │       └───────────────┘                │
+│  Client  │  └────┬──┘     │ CN=FeAcKey │            │               │                        │
+│ Frontend │       │        │            │            │       ┌───────┘                        │
+│  Access  │───────┘        └────────────┘            │       │                                │
+│    key   │                                         ┌┴───────▼───────┐                        │
+└──────────┘                                         │ Lookup Access  │    ┌─────────────────┐ │
+                                                     │  Key in Cache  │    │  Dispatch to    │ │
+                                                     │  of available  │────▶ correct Backend │ │
+                                                     │ customer pools │    └─────────────────┘ │
+                                                     └┬───────────────┘                        │
+                                                      └────────────────────────────────────────┘
+```
+###### Scope 
+For the scope of the project and simplicity of initial implementation we consider following:
+- Encryption Key RSA 3072
+- Cipher Suites: TLS v1.3 compliant set
+- Client Common Name will be pre-filled with Frontend Access Key (for active selectors)
+- TLS v1.3 as default configuration
+###### Considerations
+- mTLS Layer will match CN Access Key to the available Access Keys in the connection manager
```

Comment:
Does `Access Key` == `Access Token` ?

Reply:
`AccessToken` for `API Interface` in form of `Bearer JwtAccessToken`

`AccessKey` is bundled with `Frontend Certificate CN` for mTLS to decide routing based on certificate credentials.

I assume this is a valid routing method for this challenge, so please feel free to suggest another direction.